### PR TITLE
News refactor

### PR DESCRIPTION
**Pull request by adelq**
Objectives
- [X] Improve performance
- [X] Refactor views
- [X] Put tab bar in app bar to hide shadow
- [X] Open current url in browser
- [ ] Hiding toolbar on scroll

Bugs:
* ~~Flickering on webview~~
* Weird progress bar
* ~~Extra padding in news~~
* ~~Settings toolbar weird~~

Not ready for merge yet, but open for code reviews.

### CODE REVIEW

**Comment by adelq** [comment]
http://saulmm.github.io/mastering-coordinator/

**Comment by adelq** [comment]
@jasontsh can you take a look and let me know what you think?

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/MainActivity.java [hunk 1]
```diff
@@ -253,4 +256,26 @@ public void setNav(int id){
...
+        });
+        appBar.addView(tabLayout);
+
+        pager.setAdapter(pageAdapter);
+    }
+
+    public void removeTabs() {
+        final AppBarLayout appBar = (AppBarLayout) findViewById(R.id.appbar);
+        if (appBar != null) {
+            appBar.removeViews(1, 2);
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/MainActivity.java]: Check if this actually removes the correct view (TabLayout in AppBarLayout)

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsFragment.java [hunk 2]
```diff
@@ -84,4 +108,20 @@ public void onResume() {
...
+    public void onDestroyView() {
+        super.onDestroyView();
+        ((MainActivity) getActivity()).removeTabs();
+    }
+
+    private String getCurrentTab() {
+        try {
+            ViewFlipper flipper = (ViewFlipper) pager.getChildAt(pager.getCurrentItem());
+            WebView tab = (WebView) ((NestedScrollView) flipper.getChildAt(1)).getChildAt(0);
+            return tab.getUrl();
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsFragment.java]: Maybe find a better way of doing this?

**Comment by jasontsh** [comment]
@adelq, could you roll back one of the commits? Otherwise LGTM.  The crash fix is only a 3-lines fix.

**Pull request merged**